src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2445,6 +2444,7 @@ private bool InstructionSelfDestruct<TTracing>(EvmState vmState, ref EvmStack<TT
             ? ContractAddress.From(env.ExecutingAccount, _state.GetNonce(env.ExecutingAccount))
             : ContractAddress.From(env.ExecutingAccount, salt, initCode);
 
+        vmState.CreateList.Add(contractAddress);
```

Comment:
what would happen if the initcode fails to deploy the contract successfully? 
also is this [line](https://github.com/NethermindEth/nethermind/blob/7bbce2ffc1db1ff8278d68b81dbd040978be3a16/src/Nethermind/Nethermind.Evm/VirtualMachine.cs#L363C29-L363C72) not sufficient for this, since all successful contract depolyments will end up there ?

Reply:
no, that line is not sufficient since it happens after the initcode execution. but you are correct about one failure that i did not notice above it. moved it down now